src/verifier.js

```diff
@@ -63,10 +66,12 @@ class LocalVerifier {
   verify(req, username, password, done) {
     debug('Checking credentials', username, password);
     const query = {
-      [this.options.usernameField]: username,
       $limit: 1
     };
 
+    // set usernameField in query, accepting dot notation
+    set(query, this.options.usernameField, username);
```

Comment:
I'm wondering what the default query for nested properties should be since it depends on the database. E.g. Mongodb uses the actual dot separated string `{ query: { 'username.field': 'test' } }` instead of a nested object. 

Not sure how SQL databases (Sequelize, Knex) do do it and some (Memory, LocalStorage) do not  support nested at all. I think the safest bet might be to use the original way of setting it (`[this.options.usernameField]: username`) so it works at least with MongoDB (and for everything else you have to write your own verifier). Thoughts?